--- fam-2.7.0/src/DNotify.h.wiget	2006-02-20 13:34:22.000000000 +0100
+++ fam-2.7.0/src/DNotify.h	2006-02-20 13:34:41.000000000 +0100
@@ -59,8 +59,8 @@
     static int pipe_write_fd;
     static int pipe_read_fd;
     static int change_queue[QUEUESIZE];
-    static volatile sig_atomic_t DNotify::queue_overflowed;
-    static volatile sig_atomic_t DNotify::queue_changed;
+    static volatile sig_atomic_t queue_overflowed;
+    static volatile sig_atomic_t queue_changed;
     static volatile int queue_head; // Only modified by read handler
     static volatile int queue_tail; // Only modified by signal handler
     static EventHandler ehandler;
--- fam-2.7.0/src/Interest.h.orig	2008-08-30 13:44:30.000000000 +0200
+++ fam-2.7.0/src/Interest.h	2008-08-30 13:45:14.000000000 +0200
@@ -81,7 +81,7 @@
 protected:
 
     bool do_stat();
-    virtual void post_event(const Event&, const char * = NULL) = 0;
+    virtual void post_event(const Event&, const char * = 0) = 0;
     char& ci_bits()			{ return ci_char; }
     char& dir_bits()			{ return dir_char; }
     const char& ci_bits() const		{ return ci_char; }
