diff -Nur fam-2.6.7/fam/Makefile.am fam-2.6.7.new/fam/Makefile.am
--- fam-2.6.7/fam/Makefile.am	Tue Dec 18 02:03:37 2001
+++ fam-2.6.7.new/fam/Makefile.am	Sat Apr 13 13:14:59 2002
@@ -76,7 +76,7 @@
 
 EXTRA_fam_SOURCES = IMonIrix.c++ IMonLinux.c++ IMonNone.c++
 
-fam_LDADD = -lrpcsvc $(top_srcdir)/support/libsupport.a
+fam_LDADD = $(top_srcdir)/support/libsupport.a
 
 CLEANFILES += @MKDEPFILE@ ii_files/*.ii
 sinclude @MKDEPFILE@
