diff -uNr fam-2.6.10.orig/src/DNotify.c++ fam-2.6.10/src/DNotify.c++
--- fam-2.6.10.orig/src/DNotify.c++	2004-02-17 12:56:54.000000000 +0100
+++ fam-2.6.10/src/DNotify.c++	2004-02-17 13:00:18.321654784 +0100
@@ -37,7 +37,7 @@
 #include "Interest.h"
 #include "Log.h"
 #include "Scheduler.h"
-#include "alloc.h"
+#include <malloc.h>
 
 
 int DNotify::pipe_write_fd = -2;
diff -uNr fam-2.6.10.orig/src/IMon.c++ fam-2.6.10/src/IMon.c++
--- fam-2.6.10.orig/src/IMon.c++	2003-04-15 06:21:31.000000000 +0200
+++ fam-2.6.10/src/IMon.c++	2004-02-17 12:59:53.667402800 +0100
@@ -42,7 +42,7 @@
 #include "Interest.h"
 #include "Log.h"
 #include "Scheduler.h"
-#include "alloc.h"
+#include <malloc.h>
 
 int		   IMon::imonfd = -2;
 IMon::EventHandler IMon::ehandler = NULL;
diff -uNr fam-2.6.10.orig/lib/Client.c++ fam-2.6.10/lib/Client.c++
--- fam-2.6.10.orig/lib/Client.c++	2003-04-15 06:21:25.000000000 +0200
+++ fam-2.6.10/lib/Client.c++	2004-02-17 12:56:54.987566264 +0100
@@ -34,7 +34,8 @@
 #include <syslog.h>
 #include <errno.h>
 
-#include <iostream.h>
+#include <iostream>
+using namespace std;
 
 #include "fam.h"
 #include "Client.h"
